owncloudApp/src/main/java/com/owncloud/android/ui/activity/ShareActivity.kt

```diff
@@ -0,0 +1,394 @@
+     *
+     * @param operation Removal operation performed.
+     * @param result    Result of the removal.
+     */
+    override fun onRemoteOperationFinish(operation: RemoteOperation<*>, result: RemoteOperationResult<*>) {
+        super.onRemoteOperationFinish(operation, result)
+
+        if (result.isSuccess || operation is GetSharesForFileOperation && result.code == RemoteOperationResult.ResultCode.SHARE_NOT_FOUND) {
+            Log_OC.d(TAG, "Refreshing view on successful operation or finished refresh")
+            refreshSharesFromStorageManager()
+            if (operation is GetSharesForFileOperation) {
+                mGetSharesForFileAsyncTask = null
+            }
+        }
+
+        if (operation is CreateShareViaLinkOperation) {
+            onCreateShareViaLinkOperationFinish(operation, result as RemoteOperationResult<ShareParserResult>)
+        }
+
+        if (operation is UpdateShareViaLinkOperation) {
+            onUpdateShareViaLinkOperationFinish(operation, result as RemoteOperationResult<ShareParserResult>)
+        }
+
+        if (operation is RemoveShareOperation && result.isSuccess && editShareFragment != null) {
+            supportFragmentManager.popBackStack()
+        }
+
+        if (operation is UpdateSharePermissionsOperation
+            && editShareFragment != null && editShareFragment!!.isAdded
+        ) {
+            editShareFragment!!.onUpdateSharePermissionsFinished(result as RemoteOperationResult<ShareParserResult>?)
+        }
+    }
+
+    private fun onCreateShareViaLinkOperationFinish(
+        operation: CreateShareViaLinkOperation,
+        result: RemoteOperationResult<ShareParserResult>
+    ) {
+        if (result.isSuccess) {
+            updateFileFromDB()
+
+            publicShareFragment!!.dismiss()
+
+            fileOperationsHelper.copyOrSendPublicLink(OCShare.fromRemoteShare(result.data.shares[0]))
+
+        } else {
+            publicShareFragment!!.showError(
+                ErrorMessageAdapter.getResultMessage(result, operation, resources)
+            )
+        }
+    }
+
+    private fun onUpdateShareViaLinkOperationFinish(
+        operation: UpdateShareViaLinkOperation,
+        result: RemoteOperationResult<ShareParserResult>
+    ) {
+        if (result.isSuccess) {
+            updateFileFromDB()
+
+            publicShareFragment!!.dismiss()
+
+            fileOperationsHelper.copyOrSendPublicLink(OCShare.fromRemoteShare(result.data.shares[0]))
+
+        } else {
+            publicShareFragment!!.showError(
+                ErrorMessageAdapter.getResultMessage(result, operation, resources)
+            )
+        }
+    }
+
+    /**
+     * Updates the view, reading data from [com.owncloud.android.datamodel.FileDataStorageManager]
+     */
+    private fun refreshSharesFromStorageManager() {
+
+        val shareFileFragment = shareFileFragment
+        if (shareFileFragment != null && shareFileFragment.isAdded) {   // only if added to the view hierarchy!!
```

Comment:
`if (shareFileFragment?.isAdded == true)`

apply the same refactor for all cases in this method

Reply:
👍